apache/superset
# feat: Stop editor scrolling to top

### SUMMARY
This contribution introduces a vital usability upgrade to the SQL Editor within Apache Superset. The enhancement consists of a coding modification that embeds the cursor position into the Redux state.

Traditionally, every time users switch between SQL Lab tabs, the SQL editor viewport resets, forcing them to scroll from the top to locate their last worked-on position. This scenario presents an inconvenience that disrupts workflow and decreases editing efficiency.

This update directly addresses the issue. Upon rendering the SQL editor, it now auto-focuses on the saved cursor position from the Redux state, aligning the screen with the user's last point of work. Users can now effortlessly switch tabs, and upon their return, find the SQL Editor positioned exactly where they left off.

### BEFORE SCREENSHOTS OR ANIMATED GIF
https://github.com/apache/superset/assets/76474644/6dfaac9e-4d96-4a8c-b641-c962bf903f58

### AFTER SCREENSHOTS OR ANIMATED GIF
https://github.com/apache/superset/assets/76474644/f4430d28-9b98-4806-b85f-bab7cc70d6ab

### TESTING INSTRUCTIONS
- Navigate to the SQLLab page
- Write some query statement
- Switch to other SQL editor tab
- Switch back to the original SQL editor tab
- Verify that cursor focus on exact location where it was placed before you switched tabs

### ADDITIONAL INFORMATION
